src/main/java/org/jenkinsci/plugins/envinject/EnvInjectEnvVarsContributor.java

```diff
@@ -0,0 +1,27 @@
+package org.jenkinsci.plugins.envinject;
+
+import hudson.EnvVars;
+import hudson.Extension;
+import hudson.model.*;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+@Extension
+public class EnvInjectEnvVarsContributor extends EnvironmentContributor {
+
+    @Override
+    public void buildEnvironmentFor(Job job, EnvVars env, TaskListener listener) throws IOException, InterruptedException {
+        for (Object property : job.getProperties().values()) {
```

Comment:
@oleg-nenashev I've reverted the changes because the tests were failing. I was trying this:

```
@Override
public void buildEnvironmentFor(Job job, EnvVars env, TaskListener listener) throws IOException, InterruptedException {
    EnvInjectJobProperty jobProperty = (EnvInjectJobProperty) job.getProperty(EnvInjectJobProperty.class)
    EnvInjectJobPropertyInfo jobPropertyInfo = jobProperty.getInfo();

    // Process "Properties Content"
    Map<String, String> result = jobPropertyInfo.getPropertiesContentMap(new HashMap<String, String>());
    if (result != null) {
        env.putAll(result);
    }
}
```